### PR TITLE
Update core dependency and k8s client

### PR DESCRIPTION
**Pull request by bogdandrutu**
Lot of changes:
* All k8s.io deps updated to 0.18.6 we were using a version from 1y (or more) ago because of Prometheus dep.
* Updated code that call into k8s because some API changed
* Update semantic convention usage in aws (cc @anuraaga please check if I used correctly and most likely do a PR after to fix if I broke things) and elastic.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -108,5 +108,3 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sp
 replace github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor => ./processor/resourcedetectionprocessor/
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/processor/metricstransformprocessor => ./processor/metricstransformprocessor/
-
-replace k8s.io/client-go => k8s.io/client-go v0.0.0-20190620085101-78d2af792bab
```
Comment (bogdandrutu) [inline go.mod:112]: This was one of the hardest to find problem :(

File: extension/observer/hostobserver/extension_test.go [hunk 2]
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// +build !darwin
```
Comment (bogdandrutu) [inline extension/observer/hostobserver/extension_test.go:15]: Disabled on macos because this is not passing.

File: go.mod (hunk 1 above)
Comment (anuraaga) [reply go.mod:112]: Glad this hack is finally not needed!

**Review by anuraaga — APPROVED** [review]
Thanks! I will go in with some follow up cleanups for aws, I think they're mostly related to tests not representing real spans and not actual business logic (e.g., there's no such system as `sql` anymore.

**Review by dmitryax — APPROVED** [review]

File: extension/observer/hostobserver/extension_test.go (hunk 2 above)
Comment (bogdandrutu) [reply extension/observer/hostobserver/extension_test.go:15]: @asuresh4 I had to disable this in order to be able to run and pass tests on a macos.

File: go.mod (hunk 1 above)
Comment (bogdandrutu) [reply go.mod:112]: Unfortunately I had to add a different one, but that should be easier to remove, only need to update k8s.io/client-go > 0.19.x :)

**Pull request merged**